src/Features/Core/Portable/Diagnostics/DocumentAnalysisExecutor.cs

```diff
@@ -147,9 +150,17 @@ public async Task<IEnumerable<DiagnosticData>> ComputeDiagnosticsAsync(Diagnosti
         {
             RoslynDebug.Assert(_compilationWithAnalyzers != null);
 
-            var resultAndTelemetry = await _diagnosticAnalyzerRunner.AnalyzeDocumentAsync(analysisScope, _compilationWithAnalyzers,
-                _logPerformanceInfo, getTelemetryInfo: false, cancellationToken).ConfigureAwait(false);
-            return resultAndTelemetry.AnalysisResult;
+            try
+            {
+                var resultAndTelemetry = await _diagnosticAnalyzerRunner.AnalyzeDocumentAsync(analysisScope, _compilationWithAnalyzers,
+                    _logPerformanceInfo, getTelemetryInfo: false, cancellationToken).ConfigureAwait(false);
+                return resultAndTelemetry.AnalysisResult;
+            }
+            catch (OperationCanceledException)
```

Comment:
❔ Why do we want to execute this callback on cancellation, but _not_ execute the callback for some other exception?

Reply:
That was the only known case where we are seeing failures. I am fine with making this more generic, though I would not be able to add relevant tests.

Reply:
I would prefer to take the more conservative approach here (don't reuse on cancellation or failure). Tests for the cancellation case should be sufficient.

Reply:
Sounds good to me, let me adjust.